cirq/contrib/quirk/cells/single_qubit_rotation_cells.py

```diff
@@ -0,0 +1,110 @@
+    yield _gate("X^-⅟₁₆", ops.X**(-1 / 16))
+    yield _gate("X^-⅟₃₂", ops.X**(-1 / 32))
+    yield _gate("Y^½", ops.Y**(1 / 2))
+    yield _gate("Y^⅓", ops.Y**(1 / 3))
+    yield _gate("Y^¼", ops.Y**(1 / 4))
+    yield _gate("Y^⅛", ops.Y**(1 / 8))
+    yield _gate("Y^⅟₁₆", ops.Y**(1 / 16))
+    yield _gate("Y^⅟₃₂", ops.Y**(1 / 32))
+    yield _gate("Y^-½", ops.Y**(-1 / 2))
+    yield _gate("Y^-⅓", ops.Y**(-1 / 3))
+    yield _gate("Y^-¼", ops.Y**(-1 / 4))
+    yield _gate("Y^-⅛", ops.Y**(-1 / 8))
+    yield _gate("Y^-⅟₁₆", ops.Y**(-1 / 16))
+    yield _gate("Y^-⅟₃₂", ops.Y**(-1 / 32))
+    yield _gate("Z^½", ops.Z**(1 / 2))
+    yield _gate("Z^⅓", ops.Z**(1 / 3))
+    yield _gate("Z^¼", ops.Z**(1 / 4))
+    yield _gate("Z^⅛", ops.Z**(1 / 8))
+    yield _gate("Z^⅟₁₆", ops.Z**(1 / 16))
+    yield _gate("Z^⅟₃₂", ops.Z**(1 / 32))
+    yield _gate("Z^⅟₆₄", ops.Z**(1 / 64))
+    yield _gate("Z^⅟₁₂₈", ops.Z**(1 / 128))
+    yield _gate("Z^-½", ops.Z**(-1 / 2))
+    yield _gate("Z^-⅓", ops.Z**(-1 / 3))
+    yield _gate("Z^-¼", ops.Z**(-1 / 4))
+    yield _gate("Z^-⅛", ops.Z**(-1 / 8))
+    yield _gate("Z^-⅟₁₆", ops.Z**(-1 / 16))
+
+    # Dynamic single qubit rotations.
+    yield _gate("X^t", ops.X**sympy.Symbol('t'))
+    yield _gate("Y^t", ops.Y**sympy.Symbol('t'))
+    yield _gate("Z^t", ops.Z**sympy.Symbol('t'))
+    yield _gate("X^-t", ops.X**-sympy.Symbol('t'))
+    yield _gate("Y^-t", ops.Y**-sympy.Symbol('t'))
+    yield _gate("Z^-t", ops.Z**-sympy.Symbol('t'))
+    yield _gate("e^iXt", ops.Rx(2 * sympy.pi * sympy.Symbol('t')))
+    yield _gate("e^iYt", ops.Ry(2 * sympy.pi * sympy.Symbol('t')))
+    yield _gate("e^iZt", ops.Rz(2 * sympy.pi * sympy.Symbol('t')))
+    yield _gate("e^-iXt", ops.Rx(-2 * sympy.pi * sympy.Symbol('t')))
+    yield _gate("e^-iYt", ops.Ry(-2 * sympy.pi * sympy.Symbol('t')))
+    yield _gate("e^-iZt", ops.Rz(-2 * sympy.pi * sympy.Symbol('t')))
+
+    # Formulaic single qubit rotations.
+    yield _formula_gate("X^ft", "sin(pi*t)", lambda e: ops.X**e)
+    yield _formula_gate("Y^ft", "sin(pi*t)", lambda e: ops.Y**e)
+    yield _formula_gate("Z^ft", "sin(pi*t)", lambda e: ops.Z**e)
+    yield _formula_gate("Rxft", "pi*t*t", ops.Rx)
+    yield _formula_gate("Ryft", "pi*t*t", ops.Ry)
+    yield _formula_gate("Rzft", "pi*t*t", ops.Rz)
+
+
+def _gate(identifier: str, gate: 'cirq.Gate') -> CellMaker:
+    return CellMaker(
+        identifier=identifier,
+        size=gate.num_qubits(),
+        maker=lambda args: ExplicitOperationsCell([gate.on(*args.qubits)]))
+
+
+def _formula_gate(identifier: str, default_formula: str,
+                  gate_func: Callable[[Union[sympy.Symbol, float]], 'cirq.Gate']
+                 ) -> CellMaker:
+    return CellMaker(identifier=identifier,
+                     size=gate_func(0).num_qubits(),
+                     maker=lambda args: ExplicitOperationsCell([
+                         gate_func(
+                             parse_formula(default_formula if args.value is None
```

Comment:
It'd be good to define explicitly what we intend to support here. This would help in choosing or implementing a safe way to parse these formulae (see other comment).

Reply:
Basically the idealized goal is to exactly match what Quirk does.